packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts

```diff
@@ -0,0 +1,123 @@
+import { runtimeWalkthrough } from '../utils/functionPluginLoader';
+
+export async function createLayerWalkthrough(
+  context: any,
+  templateParameters: Partial<LayerParameters> = {},
+): Promise<Partial<LayerParameters>> {
+  _.assign(templateParameters, await inquirer.prompt(layerNameQuestion(context)));
+
+  let runtimeReturn = await runtimeWalkthrough(context, templateParameters);
+  templateParameters.runtimes = runtimeReturn.map(val => val.runtime);
+  // let { runtimePluginId } = runtimeReturn;
+
+  _.assign(templateParameters, await inquirer.prompt(layerPermissionsQuestion()));
+
+  switch (templateParameters.layerPermissions) {
+    case Permissions.awsAccs:
+      _.assign(templateParameters, await inquirer.prompt(layerAccountAccessQuestion()));
+      break;
+    case Permissions.awsOrg:
+      _.assign(templateParameters.authorizedOrgId, await inquirer.prompt(layerOrgAccessQuestion()));
+      break;
+  }
+  return templateParameters;
+}
+
+function layerNameQuestion(context: any) {
+  return [
+    {
+      type: 'input',
+      name: 'layerName',
+      message: 'Provide a name for your Lambda layer:',
+      validate: input => {
+        // TODO: make sure name is unique from other layers in project
+        if (/^[a-zA-Z0-9_\-]{1,140}$/.test(input)) {
+          return true;
+        }
+        return 'Lambda Layer names are 1-140 characters long and can only contain letters, numbers, -, _';
+      },
+      default: () => {
+        const appName = context.amplify
+          .getProjectDetails()
+          .projectConfig.projectName.toLowerCase()
+          .replace(/[^a-zA-Z0-9]/gi, '');
+        const [shortId] = uuid().split('-');
+        return `${appName}${shortId}`;
+      },
+    },
+  ];
+}
+
+function layerPermissionsQuestion() {
+  return [
+    {
+      type: 'list',
+      name: 'layerPermissions',
+      message: 'Who should have permission to use this layer?',
+      choices: [
+        {
+          name: 'Only the current AWS account',
+          value: Permissions.private,
+        },
+        {
+          name: 'Specific AWS accounts',
+          value: Permissions.awsAccs,
+        },
+        {
+          name: 'Specific AWS organization',
+          value: Permissions.awsOrg,
+        },
+        {
+          name: 'Public (everyone on AWS can use this layer)',
+          value: Permissions.public,
+        },
+      ],
+      default: Permissions.private,
+    },
+  ];
+}
+
+function layerAccountAccessQuestion() {
+  return [
+    {
+      type: 'input',
+      name: 'layerAccountAccess',
+      message: 'Provide a list of comma-separated AWS account IDs:',
+      validate: input => {
+        const accounts = input.split(',');
+        const set = new Set();
+        for (let accountID of accounts) {
+          accountID = accountID.trim();
+          if (!/^[0-9]{12}$/.test(accountID)) {
+            return 'AWS account IDs must be 12 digits long.';
+          }
+          if (set.has(accountID)) {
```

Comment:
why not just ignore the duplicate?
We could also use the [filter](https://www.npmjs.com/package/inquirer#question) option of inquirer to return the results as a Set (which would eliminate duplicates automatically)

Reply:
I considered the ignore option as well. My thought was that if you entered the same account ID twice, it was most likely a typo. I wanted to make it explicit, rather than just silently discarding. I'm open to changing this though.

Reply:
I think a warning would be better than a failure. That gives them a chance to double check the IDs without being annoying if they just happened to have a duplicate in the list

Reply:
I still think we should just warn on this rather than error. Perhaps instead of making this a condition of the inquirer validator, we could use the filter above to convert the responses to an array, then after we get the answer, read the array into a set and print out a warning if there are duplicates, but proceed with the set regardless.